core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -468,19 +479,24 @@ private[spark] class SparkNoSuchElementException(
 private[spark] class SparkSQLFeatureNotSupportedException(
     errorClass: String,
     errorSubClass: Option[String] = None,
-    messageParameters: Array[String])
+    messageParameters: Map[String, String])
```

Comment:
The exception is used ONLY once by the error class UNSUPPORTED_FEATURE.JDBC_TRANSACTION, but it is not tested at all. There is the JIRA https://issues.apache.org/jira/browse/SPARK-40391